From cc2a1f911800dc7cee51fc45bd804db04cb3ae91 Mon Sep 17 00:00:00 2001
From: Masato TOYOSHIMA <phoepsilonix@phoepsilonix.love>
Date: Fri, 5 Apr 2024 15:26:23 +0900
Subject: [PATCH] swap device is enabled.

If fdisk is used, It is also better to specify LC_ALL=C for internationalisation.
Pre-mounting swap could cause the calamares installer to fail when erasing disks or resizing partitions during installation,
but this is no longer the case with the calamares fix.
Therefore, in the live environment, automounting of swap is again enabled.
---
 PKGBUILD                  | 12 ++++++++++--
 swap-device-enabled.patch | 33 +++++++++++++++++++++++++++++++++
 2 files changed, 43 insertions(+), 2 deletions(-)
 create mode 100644 swap-device-enabled.patch

diff --git a/PKGBUILD b/PKGBUILD
index bcd57a6..1b78bc5 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -17,13 +17,21 @@ arch=('any')
 url="https://gitlab.manjaro.org/tools/development-tools/manjaro-tools-livecd"
 license=('GPL')
 makedepends=('git')
-source=("git+$url.git")
-sha256sums=('SKIP')
+source=("git+$url.git"
+        swap-device-enabled.patch
+)
+sha256sums=('SKIP'
+            '233c2fa3e5da7a6a812652a401c3c7351586ae46c9d588dc6d7631bfd443a90f')
 
 pkgver() {
 	date +%Y%m%d
 }
 
+prepare() {
+	cd ${srcdir}/${_repo} #-${pkgver}
+	patch -p1 -i ../swap-device-enabled.patch
+}
+
 build() {
 	cd ${srcdir}/${_repo} #-${pkgver}
 	make PREFIX=/usr SYSCONFDIR=/etc
diff --git a/swap-device-enabled.patch b/swap-device-enabled.patch
new file mode 100644
index 0000000..0aa5280
--- /dev/null
+++ b/swap-device-enabled.patch
@@ -0,0 +1,33 @@
+diff --git a/bin/manjaro-live.in b/bin/manjaro-live.in
+index 45e6f91..cf68871 100755
+--- a/bin/manjaro-live.in
++++ b/bin/manjaro-live.in
+@@ -27,9 +27,9 @@ echo "Got consolefont and arch $arch: $(elapsed_time_ms ${livetimer})ms" >> /var
+ ### DECIDE IF WE DON'T NEED IT ###
+ # see also https://github.com/calamares/calamares/issues/860
+ # Activate swap
+-#livetimer=$(get_timer_ms)
+-#configure_swap
+-#echo "Activated swap and added to fstab: $(elapsed_time_ms ${livetimer})ms" >> /var/log/manjaro-live.log
++livetimer=$(get_timer_ms)
++configure_swap
++echo "Activated swap and added to fstab: $(elapsed_time_ms ${livetimer})ms" >> /var/log/manjaro-live.log
+ 
+ livetimer=$(get_timer_ms)
+ configure_language
+diff --git a/lib/util-live.sh b/lib/util-live.sh
+index 8b05554..89211dd 100644
+--- a/lib/util-live.sh
++++ b/lib/util-live.sh
+@@ -309,9 +309,9 @@ configure_sudoers_d(){
+ }
+ 
+ configure_swap(){
+-	local swapdev="$(fdisk -l 2>/dev/null | grep swap | cut -d' ' -f1)"
++	local swapdev="$(lsblk -l -f -n -p | awk '{if ($2=="swap") print $1}')
+ 	if [ -e "${swapdev}" ]; then
+-		swapon ${swapdev}
++		swapon "${swapdev}" || (mkswap "${swapdev}" && swapon "${swapdev}")
+ 	fi
+ }
+ 
-- 
2.44.0

